olive/workflows/run/config.py

```diff
@@ -143,6 +158,18 @@ def validate_pass_search(cls, v, values):
         pass_cls = Pass.registry.get(v["type"].lower(), None)
         if pass_cls and pass_cls.requires_data_config():
             v["config"] = _resolve_data_config(v.get("config", {}), values, "data_config")
+
+            data_dir_name = "data_dir"
+            for config_key, config_value in v["config"].items():
+                if config_key == data_dir_name and isinstance(config_value, dict):
+                    rp_type = config_value.get("type")
+                    if rp_type in AZUREML_RESOURCE_TYPES:
+                        rp_aml_client = config_value.get("config", {}).get("azureml_client")
+                        if not rp_aml_client:
+                            if "azureml_client" not in values:
+                                raise ValueError("azureml_client is required for azureml resource path in config")
+                            config_value["config"]["azureml_client"] = values["azureml_client"]
```

Comment:
it seems the logic is exactly same with above one? could we reuse it by using a method?

Reply:
Updated.